src/content/docs/guides/test/mocking.md

```diff
@@ -0,0 +1,171 @@
+    value: {
+      // @ts-ignore      
+      getRandomValues: (buffer) => {
+        return randomFillSync(buffer);
+      },
+    },
+  });
+});
+
+
+test("invoke", async () => {
+  mockIPC((cmd, args) => {
+    // simulated rust command called "add" that just adds two numbers
+    if(cmd === "add") {
+      return (args.a as number) + (args.b as number);
+    }
+  });
+
+  // we can use the spying tools provided by vitest to track the mocked function
+  const spy = vi.spyOn(window, "__TAURI_IPC__");
+
+  expect(invoke("add", { a: 12, b: 15 })).resolves.toBe(27);
+  expect(spy).toHaveBeenCalled();
+});
+```
+
+To mock IPC requests to a sidecar or shell command you need to grab the ID of the event handler when `spawn()` or `execute()` is called and use this ID to emit events the backend would send back:
+
+```js
+mockIPC(async (cmd, args) => {
+  if (args.message.cmd === 'execute') {
+    const eventCallbackId = `_${args.message.onEventFn}`;
+    const eventEmitter = window[eventCallbackId];
+
+    // 'Stdout' event can be called multiple times
+    eventEmitter({
+      event: 'Stdout',
+      payload: 'some data sent from the process',
+    });
+
+    // 'Terminated' event must be called at the end to resolve the promise
+    eventEmitter({
+      event: 'Terminated',
+      payload: {
+        code: 0,
+        signal: 'kill',
+      },
+    });
+  }
+});
+```
+
+## Windows
+
+Sometimes you have window-specific code (a splash screen window, for example), so you need to simulate different windows.
+You can use the [`mockWindows()`] method to create fake window labels. The first string identifies the "current" window (i.e., the window your JavaScript believes itself in), and all other strings are treated as additional windows.
+
+:::note
+
+[`mockWindows()`] only fakes the existence of windows but no window properties. To simulate window properties, you need to intercept the correct calls using [`mockIPC()`]
+
+:::
+
+```js
+import { beforeAll, expect, test } from 'vitest';
+import { randomFillSync } from 'crypto';
+
+import { mockWindows } from '@tauri-apps/api/mocks';
+
+// jsdom doesn't come with a WebCrypto implementation
+beforeAll(() => {
+  Object.defineProperty(window, 'crypto', {
+    value: {
+      // @ts-ignore      
+      getRandomValues: (buffer) => {
+        return randomFillSync(buffer);
+      },
+    },
+  });
+});
+
+test('invoke', async () => {
+  mockWindows('main', 'second', 'third');
+
+  const { getCurrent, getAll } = await import('@tauri-apps/api/window');
+
+  expect(getCurrent()).toHaveProperty('label', 'main');
+  expect(getAll().map((w) => w.label)).toEqual(['main', 'second', 'third']);
+});
+```
+
+[`@tauri-apps/api/mocks`]: https://tauri.app/v1/api/js/mocks/
+[`mockipc()`]: https://tauri.app/v1/api/js/mocks#mockipc
+[`mockwindows()`]: https://tauri.app/v1/api/js/mocks#mockwindows
+[`clearmocks()`]: https://tauri.app/v1/api/js/mocks#clearmocks
+[vitest]: https://vitest.dev
```

Comment:
links to v1 api docs, after a two seconds quick search I didn't find on v2

Reply:
Good catch, asked in Discord https://discord.com/channels/616186924390023171/986184094050316358/1184092673133522996